dataRetriever/storageRequesters/equivalentProofsRequester.go

```diff
@@ -0,0 +1,147 @@
+package storagerequesters
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data/endProcess"
+	"github.com/multiversx/mx-chain-core-go/data/typeConverters"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+	"github.com/multiversx/mx-chain-go/dataRetriever/resolvers/epochproviders/disabled"
+	"github.com/multiversx/mx-chain-go/storage"
+	"sync"
+	"time"
+)
+
+// ArgEquivalentProofsRequester is the argument structure used to create a new equivalent proofs requester instance
+type ArgEquivalentProofsRequester struct {
+	Messenger                dataRetriever.MessageHandler
+	ResponseTopicName        string
+	ManualEpochStartNotifier dataRetriever.ManualEpochStartNotifier
+	ChanGracefullyClose      chan endProcess.ArgEndProcess
+	DelayBeforeGracefulClose time.Duration
+	NonceConverter           typeConverters.Uint64ByteSliceConverter
+	Storage                  dataRetriever.StorageService
+	Marshaller               marshal.Marshalizer
+}
+
+type equivalentProofsRequester struct {
+	*storageRequester
+	nonceConverter  typeConverters.Uint64ByteSliceConverter
+	storage         dataRetriever.StorageService
+	marshaller      marshal.Marshalizer
+	mutEpochHandler sync.RWMutex
+	epochHandler    dataRetriever.EpochHandler
+}
+
+// NewEquivalentProofsRequester returns a new instance of equivalent proofs requester
+func NewEquivalentProofsRequester(args ArgEquivalentProofsRequester) (*equivalentProofsRequester, error) {
+	err := checkArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	return &equivalentProofsRequester{
+		storageRequester: &storageRequester{
+			messenger:                args.Messenger,
+			responseTopicName:        args.ResponseTopicName,
+			manualEpochStartNotifier: args.ManualEpochStartNotifier,
+			chanGracefullyClose:      args.ChanGracefullyClose,
+			delayBeforeGracefulClose: args.DelayBeforeGracefulClose,
+		},
+		nonceConverter: args.NonceConverter,
+		storage:        args.Storage,
+		marshaller:     args.Marshaller,
+		epochHandler:   disabled.NewEpochHandler(),
+	}, nil
+}
+
+func checkArgs(args ArgEquivalentProofsRequester) error {
+	if check.IfNil(args.Messenger) {
+		return dataRetriever.ErrNilMessenger
+	}
+	if check.IfNil(args.ManualEpochStartNotifier) {
+		return dataRetriever.ErrNilManualEpochStartNotifier
+	}
+	if args.ChanGracefullyClose == nil {
+		return dataRetriever.ErrNilGracefullyCloseChannel
+	}
+	if check.IfNil(args.NonceConverter) {
+		return dataRetriever.ErrNilUint64ByteSliceConverter
+	}
+	if check.IfNil(args.Storage) {
+		return dataRetriever.ErrNilStore
+	}
+	if check.IfNil(args.Marshaller) {
+		return dataRetriever.ErrNilMarshalizer
+	}
+
+	return nil
+}
+
+func (requester *equivalentProofsRequester) RequestDataFromHash(hashShardKey []byte, epoch uint32) error {
```

Comment:
doesn't seem like you are using the epoch here, maybe ignore the argument.

Reply:
indeed, removed